src/osd/PGBackend.cc

```diff
@@ -447,7 +447,7 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
 		  << (known ? "known" : "best guess")
 		  << " omap_digest 0x" << auth.omap_digest << std::dec
 		  << " from auth shard " << auth_shard;
-      result.set_omap_digest_mismatch();
+      result.set_omap_digest_mismatch(known);
```

Comment:
Ok, so the guessed field on each shard is or'd with the global object errors to build up the global object errors.  I don't think "guessed" is a good name.  We don't know which shard is wrong, but we aren't guessing that there is a mismatch.  Also, is the only difference between the case where we have no oi digest but the shards to not match and the case where we do have a the oi.digest and the shards do not match that in the latter case at least one shard will have have a digest error?